packages/backend/src/database/migrations/20240515151028_add_custom_dimension_to_selected_dimensions.ts

```diff
@@ -0,0 +1,51 @@
+import { DBFieldTypes } from '@lightdash/common';
+import { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+    const customBinDimensions = await knex(
+        'saved_queries_version_custom_dimensions',
+    ).select('saved_queries_version_id', 'id');
+    const customSqlDimensions = await knex(
+        'saved_queries_version_custom_sql_dimensions',
+    ).select('saved_queries_version_id', 'id');
+
+    const insertPromises = [...customBinDimensions, ...customSqlDimensions].map(
+        (customDimension) =>
+            knex('saved_queries_version_fields').insert({
+                saved_queries_version_id:
+                    customDimension.saved_queries_version_id,
+                name: customDimension.id,
+                field_type: DBFieldTypes.DIMENSION,
+                order: 99,
+            }),
+    );
+    console.debug(
+        `Selecting ${insertPromises.length} custom dimensions in saved charts`,
+    );
+    await Promise.all(insertPromises);
+}
+
+export async function down(knex: Knex): Promise<void> {
+    const customBinDimensions = await knex(
+        'saved_queries_version_custom_dimensions',
+    ).select('saved_queries_version_id', 'id');
+    const customSqlDimensions = await knex(
+        'saved_queries_version_custom_sql_dimensions',
+    ).select('saved_queries_version_id', 'id');
+
+    const deletePromises = [...customBinDimensions, ...customSqlDimensions].map(
+        (customDimension) =>
+            knex('saved_queries_version_fields')
+                .delete()
+                .where('name', customDimension.id)
+                .andWhere(
+                    'saved_queries_version_id',
+                    customDimension.saved_queries_version_id,
+                ),
+    );
```

Comment:
A similar approach with 1 query by moving the loop inside the query builder. If I'm correct this will build a query like:

`delete where (id A and name A) or (id B and name B) ...`

```suggestion
 knex('saved_queries_version_fields')
   .delete()
   .where((builder) => {
        [
        ...customBinDimensions,
        ...customSqlDimensions
        ].forEach((customDimension) => {
                builder.orWhere((nestedBuilder) => {
                    nestedBuilder
                      .where('name', customDimension.id)
                      .andWhere(
                          'saved_queries_version_id',
                          customDimension.saved_queries_version_id,
                      ),
                })

        })
   })
```